### PR TITLE
[JBPM-9625] Allow to bypass the authenticated user in the KIE REST API that is adding users or groups as potential owners

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9625
depends on https://github.com/kiegroup/jbpm/pull/1873

### CODE REVIEW

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java [hunk 1]
```diff
@@ -55,179 +55,286 @@ public UserTaskAdminServicesClientImpl(KieServicesConfiguration config, ClassLoa
...
+            if(BYPASS_AUTH_USER && userId != null) {
+                queryString += "&user=" + userId;
+            }
 
             makeHttpPutRequestAndCreateCustomResponse(
                     build(loadBalancer.getUrl(), ADMIN_TASK_URI + "/" + TASK_INSTANCE_POT_OWNERS_USERS_URI, valuesMap) + queryString, orgEntities, null, headers);
         } else {
+            Object[] argvs = new Object[]{containerId, taskId, removeExisting};
+            if(BYPASS_AUTH_USER && userId != null) {
+                argvs = new Object[]{userId, containerId, taskId, removeExisting};
```
Comment (sutaakar) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java:79]: JMS messages sent by client are converted to appropriate base classes, for example for `UserTaskAdminService` there is triggered `UserTaskAdminServiceBase` instance. This instance has only a method containing `userId` parameter. So some value needs to be provided as `userId` for the JMS in all cases.

Same for other cases below.

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java [hunk 2]
```diff
@@ -55,179 +55,286 @@ public UserTaskAdminServicesClientImpl(KieServicesConfiguration config, ClassLoa
...
+    @Override
+    public void removePotentialOwnerUsers(String userId, String containerId, Long taskId, String... users) {
 
         if( config.isRest() ) {
             Map<String, Object> valuesMap = new HashMap<String, Object>();
             valuesMap.put(CONTAINER_ID, containerId);
             valuesMap.put(TASK_INSTANCE_ID, taskId);
             valuesMap.put(ENTITY_ID, Arrays.stream(users).collect(Collectors.joining(",")));
 
+            String queryString = "";
```
Comment (sutaakar) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java:163]: https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/AbstractKieServicesClientImpl.java#L696 can be reused here

**Comment by sutaakar** [comment]
@elguardian Could you provide some test coverage, or should I take care of it?

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java:79]: the other way round... old calls need to come back for backward compatibility.

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/UserTaskAdminServiceBase.java [hunk 3]
```diff
@@ -62,68 +63,127 @@
...
+        } else {
+            userTaskAdminService.addBusinessAdmins(userId, containerId, taskId.longValue(), removeExisting, entities);
+        }
+
+
         logger.debug("Business admins {} added to task {}", entities, taskId);
     }
 
     public void removePotentialOwners(String containerId, Number taskId, List<String> orgEntities, boolean isUser) {
+        removePotentialOwners(null, taskId, orgEntities, isUser);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/UserTaskAdminServiceBase.java:133]:
```suggestion
        removePotentialOwners(null, containerId, taskId, orgEntities, isUser);
```

**Comment by sutaakar** [comment]
@elguardian Do you plan to provide some additional test coverage, or should I take care of it?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/UserTaskAdminServiceBase.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/UserTaskAdminServiceBase.java]: done

**Comment by elguardian** [comment]
@sutaakar I would appreciate if you take care of it. thx

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
Jenkins retest this

**Comment by sutaakar** [comment]
@elguardian Test coverage prepared as https://github.com/elguardian/droolsjbpm-integration/pull/5
Feel free to add it to your branch, or can be done as separate PR here.

**Pull request merged**